lib/pbench/cli/agent/commands/generate_token.py

```diff
@@ -0,0 +1,71 @@
+"""pbench-generate-token"""
+
+import click
+import json
+import requests
+
+from pbench.agent.base import BaseCommand
+from pbench.cli.agent import pass_cli_context
+from pbench.cli.agent.options import common_options
+
+
+class GenerateToken(BaseCommand):
+    def __init__(self, context):
+        super().__init__(context)
+
+    def execute(self):
+        """Generate a token used to access the pbench server RESTful API"""
+
+        # FIXME:  Should we be logging anything here?
+        #
+        #  self.logger.warn(
+        #     "%s deprecated, will be removed in future release in favor of pbench-clear-results",
+        #     self.name,
+        # )
+
+        server = self.config.agent.parser.get("results", "server_rest_url")
+        headers = {"Content-Type": "application/json", "Authorization": None}
+        payload = {
+            "username": self.context.username,
+            "password": self.context.password,
+            "token_duration": self.context.token_duration
+        }
+        try:
+            response = requests.post(
+                f"{server}/login", headers=headers, json=payload
+            )
+        except requests.exceptions.ConnectionError as exc:
+            raise exc.args[0].reason
+
+        payload = json.loads(response.text)
+        if response.status_code == 200:
+            click.echo(payload["auth_token"])
+            return 0
+
+        click.echo(payload["message"], err=True)
+        return 1
+
+
+@click.command()
+@common_options
+@click.option('--username', prompt=True, required=True,
+              help='pbench server account username '
+                   '(will prompt if unspecified)')
+@click.option('--password', prompt=True, hide_input=True, required=True,
+              help='pbench server account password '
+                   '(will prompt if unspecified)')
+@click.option('--token-duration', type=int, required=False, default=3600,
```

Comment:
As things stand right now, the token duration is fixated on the server side and we specify it in the server config file

Reply:
Agreed.  (When the time is right, we can consider changing that, but I don't think we need to until we have some sort of roles implementation.)